sdk/src/client/api/address.rs

```diff
@@ -156,27 +157,35 @@ impl<'a> GetAddressesBuilder<'a> {
 
     /// Consume the builder and get the vector of public and internal addresses bech32 encoded
     pub async fn get_all(self) -> Result<Bech32Addresses> {
-        let bech32_hrp = match self.bech32_hrp.clone() {
-            Some(bech32_hrp) => bech32_hrp,
+        let bech32_hrp: Cow<'_, str> = match &self.bech32_hrp {
+            Some(bech32_hrp) => bech32_hrp.as_str().into(),
             None => match self.client {
-                Some(client) => client.get_bech32_hrp().await?,
-                None => SHIMMER_TESTNET_BECH32_HRP.to_string(),
+                Some(client) => client.get_bech32_hrp().await?.into(),
+                None => SHIMMER_TESTNET_BECH32_HRP.into(),
             },
         };
-        let addresses = self.get_all_raw().await?;
+        let addresses = self.get_all_internal().await?;
 
         Ok(Bech32Addresses {
-            public: addresses.public.into_iter().map(|a| a.to_bech32(&bech32_hrp)).collect(),
+            public: addresses
+                .public
+                .into_iter()
+                .map(|a| Ok(Bech32Address::new(bech32_hrp.as_ref(), a)?))
+                .collect::<Result<_>>()?,
             internal: addresses
                 .internal
                 .into_iter()
-                .map(|a| a.to_bech32(&bech32_hrp))
-                .collect(),
+                .map(|a| Ok(Bech32Address::new(bech32_hrp.as_ref(), a)?))
+                .collect::<Result<_>>()?,
         })
     }
 
     /// Consume the builder and get the vector of public and internal addresses
     pub async fn get_all_raw(self) -> Result<RawAddresses> {
+        self.get_all_internal().await
+    }
+
+    async fn get_all_internal(&self) -> Result<RawAddresses> {
```

Comment:
Why this?

Reply:
Because this fn does not consume `self` and can be used with the borrowing impl above.